### PR TITLE
US021: View Reservation History List

### PR DESCRIPTION
**Pull request by SamDotNetDev**

### CODE REVIEW

**Review by Husniddin1908 — APPROVED** [review]

File: ParkingZoneApp/Areas/User/Controllers/ReservationController.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using ParkingZoneApp.Services;
+using ParkingZoneApp.ViewModels.ReservationVMs;
+using System.Security.Claims;
+
+namespace ParkingZoneApp.Areas.User.Controllers
+{
+    [Area("User")]
+    [Authorize(Roles = "User")]
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/Areas/User/Controllers/ReservationController.cs]: in the task you are not required to add roles! remove roles

File: ParkingZoneApp/Areas/User/Views/Reservation/Index.cshtml [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        touch-action: manipulation;
+        width: fit-content;
+        word-break: break-word;
+        border: 0;
+    }
+</style>
+
+@if(Model.Count() == 0)
+{
+    <h1>Reservation histories do not exists</h1>
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/Areas/User/Views/Reservation/Index.cshtml:69]:
```suggestion
    <h1>Reservation histories do not exist</h1>
```

Comment (abdullokhtolibjonov) [reply ParkingZoneApp/Areas/User/Views/Reservation/Index.cshtml:69]: grammar

File: ParkingZoneApp/Areas/User/Views/Reservation/Index.cshtml [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                    <td>
+                        @Html.DisplayFor(modelItem => item.VehicleNumber)
+                    </td>
+                    <td>
+                        <div>
+                            @if(item.IsActive)
+                            {
+                                <a asp-action="Prolong" asp-route-id="@item.Id" class="btn btn-outline-dark"> Prolong </a>
+                            }
+                        </div>
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/Areas/User/Views/Reservation/Index.cshtml]: please do not add not required things

File: ParkingZoneApp/Models/Reservation.cs [hunk 4]
```diff
@@ -29,5 +29,11 @@ public class Reservation
 
         [Required]
         public virtual ApplicationUser User { get; set; }
+
+        [NotMapped]
+        public bool IsActive 
+        { 
+            get => StartTime <= DateTime.Now && StartTime.AddHours(Duration) > DateTime.Now;
+        }
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/Models/Reservation.cs:37]: why did you added it? where did you used it?

File: ParkingZoneApp/Program.cs [hunk 5]
```diff
@@ -13,13 +14,14 @@ public static void Main(string[] args)
...
             // Add services to the container.
-            var connectionString = builder.Configuration.GetConnectionString("DefaultConnection") 
+            var connectionString = builder.Configuration.GetConnectionString("DefaultConnection")
                 ?? throw new InvalidOperationException("Connection string 'DefaultConnection' not found.");
             builder.Services.AddDbContext<ApplicationDbContext>(options =>
                 options.UseSqlServer(connectionString));
             builder.Services.AddDatabaseDeveloperPageExceptionFilter();
 
             builder.Services.AddDefaultIdentity<ApplicationUser>(options => options.SignIn.RequireConfirmedAccount = false)
+                .AddRoles<IdentityRole>()
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/Program.cs]: remove roles

**Review by abdullokhtolibjonov — CHANGES_REQUESTED** [review]

File: ParkingZoneApp/Models/Reservation.cs (hunk 4 above)
Comment (SamDotNetDev) [reply ParkingZoneApp/Models/Reservation.cs:37]: it is for html i check whether is it active if it's so I'll make the background green as written in to do list

File: ParkingZoneApp/ViewModels/ReservationVMs/ReservationHistoryListVM.cs [hunk 6]
```diff
@@ -0,0 +1,46 @@
+﻿using ParkingZoneApp.Models;
+using System.ComponentModel.DataAnnotations;
+
+namespace ParkingZoneApp.ViewModels.ReservationVMs
+{
+    public class ReservationHistoryListVM
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/ViewModels/ReservationVMs/ReservationHistoryListVM.cs]:
```suggestion
    public class ReservationHistoryListItemVM
```

File: ParkingZoneApp/Areas/Admin/Controllers/ParkingZoneController.cs [hunk 7]
```diff
@@ -6,7 +6,6 @@
 
 namespace ParkingZoneApp.Areas.Admin
 {
-    [Authorize]
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/Areas/Admin/Controllers/ParkingZoneController.cs:9]: why did you removed that? dont delete write again. dont change parking zone you task is not it

File: ParkingZoneApp/Areas/Admin/Controllers/ParkingSlotController.cs [hunk 8]
```diff
@@ -5,7 +5,6 @@
 
 namespace ParkingZoneApp.Areas.Admin
 {
-    [Authorize]
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/Areas/Admin/Controllers/ParkingSlotController.cs:8]: dont delete

**Review by abdullokhtolibjonov — CHANGES_REQUESTED** [review]

**Review by abdullokhtolibjonov — APPROVED** [review]

**Pull request merged**